apps/backend/src/location/location.service.ts

```diff
@@ -14,36 +13,61 @@ export class LocationService {
 
   public async getLocationsByID(
     program: Ministries,
-    location_ids: number[],
-    method: LocationMethod
-  ): Promise<LocationEntity[]> {
-    const locationMethod =
-      method === LocationMethod.Bank
-        ? LocationMethod.Bank
-        : Not(LocationMethod.Bank);
-    return await this.locationRepo.find({
+    location_ids: number[]
+  ): Promise<NormalizedLocation[]> {
+    const locations = await this.locationRepo.find({
       where: {
         source_id: program,
-        method: locationMethod,
         location_id: In(location_ids),
       },
       order: {
         location_id: 'ASC',
       },
     });
+    return this.normalizeLocations(locations);
+  }
+  public normalizeLocations(locations: LocationEntity[]): NormalizedLocation[] {
+    const normalizedLocationList = locations.reduce(
+      (acc: { [key: string]: NormalizedLocation }, itm: LocationEntity) => {
+        const key = itm.location_id;
+        if (!acc[key]) {
+          acc[key] = {
+            source_id: itm.source_id,
+            location_id: itm.location_id,
+            program_code: itm.program_code,
+            ministry_client: itm.ministry_client,
+            resp_code: itm.resp_code,
+            service_line_code: itm.service_line_code,
+            stob_code: itm.stob_code,
+            project_code: itm.project_code,
+            merchant_ids: [],
+            pt_location_id: 0,
+            description: '',
+          };
+        }
+        itm.merchant_id !== 999999999 &&
```

Comment:
I know why because you explained it to me, but this should be commented on why. We may also want to define `999999999` as a constant. We always want to avoid magic numbers